### ISSUE TITLE
Decompiler does not remove transparent identifier expression on root level

### ISSUE THREAD

**Issue by AradAral** [hazzik/DelegateDecompiler#167]
Hi again! I'm encountering another bug.
I'm trying to decompile this method:

```csharp
public static CommentDto ToCommentDto(this Comment comment)
{
    return new CommentDto
    {
        IsUserPurchaser = comment.Course.PurchaseItems.Any(pi => pi.Purchase.UserId == comment.UserId && pi.Purchase.IsFinalized),
    };
}
```

But surprisingly DelegateDecompiler gives me this:
```csharp
comment => new CommentDto()
{
    IsUserPurchaser = new { comment = comment }.comment.Course.PurchaseItems.Any(pi => ((pi.Purchase.UserId == comment.UserId) && pi.Purchase.IsFinalized))
};
```

It's strange! Where is `new { comment = comment }` even coming from?!

**Comment by hazzik** [comment]
> Where is new { comment = comment } even coming from?!

It is from the closure. Need a visitor to unwrap such closures.

**Comment by hazzik** [comment]
Should be fixed in 0.29.0
